lib/cfndsl/metadata.rb

```diff
@@ -0,0 +1,24 @@
+module CfnDsl
+  # Handles metadata objects for AWS launch configurations
+  #
+  # Usage
+  # Metadata("AWS::CloudFormation::Init", {
+  # "config" => {
+  #  "files"    => {
+  #    "/cfn.ini" => {
+  #      "content" => "test"
+  #     }
+  #    }
+  #  }
+  # })
+  #
+  class MetadataDefinition < JSONable
```

Comment:
One final comment: how did this slip into your change? 
Metadata is defined for resources here: https://github.com/stevenjack/cfndsl/blob/master/lib/cfndsl/resources.rb#L8
and for templates here: https://github.com/stevenjack/cfndsl/blob/master/lib/cfndsl/orchestration_template.rb#L11
I know some issues have cropped up recently regarding the dsl_attrs, but this PR is not the place to address them. Maybe this wasn't intended to be included?